src/lazy/any_encoding.rs

```diff
@@ -599,20 +600,20 @@ impl<'data> LazyRawReader<'data, AnyEncoding> for LazyRawAnyReader<'data> {
         }
     }
 
-    fn stream_data(&self) -> (&'data [u8], usize, IonEncoding) {
+    fn save_state(&self) -> RawReaderState<'data> {
         use RawReaderKind::*;
-        let (remaining_data, stream_offset, mut encoding) = match &self.encoding_reader {
-            Text_1_0(r) => r.stream_data(),
-            Binary_1_0(r) => r.stream_data(),
-            Text_1_1(r) => r.stream_data(),
-            Binary_1_1(r) => r.stream_data(),
+        let reader_state = match &self.encoding_reader {
+            Text_1_0(r) => r.save_state(),
+            Binary_1_0(r) => r.save_state(),
+            Text_1_1(r) => r.save_state(),
+            Binary_1_1(r) => r.save_state(),
```

Comment:
It took me a while to find the comment: https://github.com/amazon-ion/ion-rust/pull/796#discussion_r1672742255

> Doc comment would be helpful because it's not immediately obvious whether "stream" is a verb or a noun here.

It's a little ironic because "save state" is just as ambiguous as "stream data". Both "save" and "stream" may be adjectives or verbs.

Reply:
If it's a noun, consider `data_stream`, `saved_state`. If it's a verb, consider `save_reader_state` maybe?

Or even `get_reader_state()`?

Reply:
While the name `stream_data` was noun/verb ambiguous, I think the actual problem being solved was that the name didn't explain what it did. I think `save_state` addresses that, even if it remains noun/verb ambiguous.